### ISSUE TITLE
no right click on tray items

### ISSUE THREAD

**Issue by oberon2007** [linuxdeepin/dde-dock#14]
When I have at least two entries in the tray I can click on the tray to get a pop-up selection of the contained entries and can then either right or left-click them. But the tray apparently does not listen to right-click action. As long as it is something like **skype**, although I lose the associated right-click actions I can at least left-click to open the application, but with for example **pamac-tray** this makes the tray entry completely inaccessible, because it has no left-click action associated.
![tray](https://cloud.githubusercontent.com/assets/11800714/17810556/23b49864-661e-11e6-862e-b55d9ffa4db1.jpg)
Also mouseover does not work on the tray-entries.

**Comment by sbwtw** [comment]
it's feature

**Comment by oberon2007** [comment]
It's a problem if it makes the tray items un-usable! How can that be a feature? :wink:

**Comment by oberon2007** [comment]
Maybe you'd like to consider at least that the popup of contained items with right-click also happens when there is only one item present? I think the current version is just not acceptable. As I say, some tray items become completely unaccessible, others will lose some of there functionality until at least another item is present.

**Comment by denseleznev** [comment]
I can confirm the bug, I have only one icon of network-manager in my system tray, and it is completely unclickable with both mouse buttons.

**Comment by guanghwang** [comment]
I am able to right click it if there is only one icon in Deepin 2015.3. The version installed in my system is 
```
dde-dock:
  Installed: 4.0.1-1
```.
![deepinscreenshot20160824092926](https://cloud.githubusercontent.com/assets/12839553/17932350/9571f8a8-69dd-11e6-9931-862102cbe929.png)

**Comment by sbwtw** [comment]
@guanghwang if you have more than 1 systray, you need to left-cllick this circle to show popup window, and do your operate at popup window

@oberon2007 @denseleznev dde-dock have new version released, please test new version

**Comment by oberon2007** [comment]
@sbwtw theoretically this makes me really happy. It works for **skype** ok. But not for our [**pamac-tray**](https://github.com/manjaro/pamac), which is a gtk3 app. Maybe that's why? I will keep trying with different applications to determine what the difference could be.

**Comment by oberon2007** [comment]
No luck with [**nm-applet**](https://www.archlinux.org/packages/extra/x86_64/network-manager-applet/) either.

**Comment by sbwtw** [comment]
i have confirmed nm-applet is not work.

**Comment by sbwtw** [comment]
seems i send fake click event to application tray window, and gtk3 app will reject fake events. that's a annoying problem (

**Comment by sbwtw** [comment]
@oberon2007 hi, whats your network-manager-gnome version ?
my version is 1.1.91~3deepin in deepin mirror.
but my friend test v1.2.2 in debian mirror is work.

**Comment by oberon2007** [comment]
This: https://www.archlinux.org/packages/testing/x86_64/network-manager-applet/ :)

**Comment by oberon2007** [comment]
I know **nm-applet** is not needed in Deepin, it's just an example for a tray-app that is not working here. The crucial one for us in Manjaro is **pamac**, our package manager. We need it! :wink:
[sources](https://github.com/manjaro/pamac/)
[PKGBUILD](https://github.com/manjaro/packages-extra/blob/master/pamac/PKGBUILD)
[package](http://repo.manjaro.org.uk/pool/overlay/pamac-4.1.2-1-x86_64.pkg.tar.xz) (should normally be possible to install it on Arch if it's just to evaluate the tray behaviour)

**Comment by sbwtw** [comment]
@oberon2007  https://github.com/linuxdeepin/dde-dock/commit/2584e8222672986cde558ec103165031fd49bee9 this commit can solve this problem, i have test nm-applet is worked.

**Comment by oberon2007** [comment]
Thank you, @sbwtw . For me it doesn't work properly, yet, I'm afraid... Using this commit, the dock does something funny now:
When I right click **nm-applet**, nothing happens, but when I then move the mouse away from the dock, this is interpreted as a right click on the desktop.

![](http://imgur.com/ZKVzzxpl.png)

Or when I move it slowly it will bring up the dock settings.
Also now the auto-unhide is broken, meaning when a window is maximized the dock will not unhide when I mouse over it.

I'm not sure what you mean with
> need xdotool command

? So you mean a new dependency? **xdotool** is installed. Anything else? Am I missing something?

**Comment by sbwtw** [comment]
@oberon2007  :(, im not sure why doesnt work at your platform but its worked at deepin. im optimize some code, use XTEST extension to send fake event instead of xdotool. you can try to test last commit, . and need to add new denpendencies **libxtst** 

about your new problem, the dock cant auto unhide, is my fault and its fixed completely at this commit https://github.com/linuxdeepin/dde-dock/commit/d16c92a94ba1bf19255659bc21445869c0dddd35

**Comment by oberon2007** [comment]
Hi @sbwtw It's better and worse now :laughing:
auto-unhide is working again and I can click a single item in the tray, but the problem is that **dde-dock** crashes immediately with 
``symbol lookup error: /usr/lib/dde-dock/plugins/libsystem-tray.so: undefined symbol: XTestFakeMotionEvent``
Am I maybe missing a newer version of another package? I have added **libxtst** as a depends...

**Comment by sbwtw** [comment]
@oberon2007 libxtst is a X11 library, not a dde-package. and my package version is **v1.2.2** too. the `XTestFakeMotionEvent` is defined at `/usr/include/X11/extensions/XTest.h`. i dont know why it crash at your environment, maybe dde-dock or plugins to linked to libxtst library. you can try to execute:
`ldd /usr/lib/dde-dock/plugins/libsystem-tray.so /usr/bin/dde-dock | grep -i xtst`
and output should be like this:
`libXtst.so.6 => /usr/lib/x86_64-linux-gnu/libXtst.so.6 (0x00007f9dd3d8b000)`

**Comment by snyh** [comment]
all  x11 extension must be queried before send any other requests!

**Comment by oberon2007** [comment]
Not sure why, but in any case this is fixed for me with **dde-dock 4.0.2** in combination with all the other latest pkg-releases :smile: 
Thank you!
